File: python/3111. Minimum Rectangles to Cover Points.py
class Solution:
    def minRectanglesToCoverPoints(self, points: List[List[int]], w: int) -> int:
        points.sort(key=lambda x: x[0])
        previous = -1
        answer = 0
        for point in points:
            if point[0] > previous:
                answer += 1
                previous = point[0] + w
        return answer

File: go/3111. Minimum Rectangles to Cover Points.go
func minRectanglesToCoverPoints(points [][]int, w int) int {
	sort.Slice(points, func(i, j int) bool {
		return points[i][0] < points[j][0]
	})
	answer := 0
	previous := -1
	for _, p := range points {
		if p[0] > previous {
			answer += 1
			previous = p[0] + w
		}
	}
	return answer
}